client/src/game/gameObjects/moveableGameObject.ts

```diff
@@ -25,36 +25,35 @@ class MoveableGameObject extends GameObject implements IMoveableGameObject {
 				}
 			}
 		}
-
 		super.update(time, delta)
 		// updateData('position')
 	}
 
 	private move(source: ICordinates, destination: ICordinates, delta: number): ICordinates {
-		return {
-			x: this.getDistanceToMove(source.x, destination.x, delta),
-			y: this.getDistanceToMove(source.y, destination.y, delta),
-			z: this.getDistanceToMove(source.z, destination.z, delta)
+		const direction = {
+			x: destination.x - source.x,
+			y: destination.y - source.y,
+			z: destination.z - source.z
 		}
-	}
-
-	private getDistanceToMove = (source: number, destination: number, delta: number): number => {
-		let newPostion: number = source
-		let moveDistance = delta * this.movementSpeed
-		const diff = Math.abs(destination - source)
-		if (diff === 0) {
-			return destination
+		const posToDesLength = Math.sqrt(direction.x * direction.x + direction.y * direction.y + direction.z * direction.z)
+		if (posToDesLength === 0) {
+			return source
 		}
-		if (diff < moveDistance) {
-			moveDistance = diff
+		const velocity = {
+			x: (direction.x / posToDesLength) * delta * this.movementSpeed,
+			y: (direction.y / posToDesLength) * delta * this.movementSpeed,
+			z: (direction.z / posToDesLength) * delta * this.movementSpeed
 		}
-		if (source > destination) {
-			newPostion = source - moveDistance
+		const velocityMagnitude = Math.sqrt(velocity.x * velocity.x + velocity.y * velocity.y + velocity.z * velocity.z)
+		if (posToDesLength < velocityMagnitude) {
+			return destination
 		}
-		if (source < destination) {
-			newPostion = source + moveDistance
+
+		return {
```

Comment:
Would it make sense to put the returned object literal into a const with a sensible name, so it'd be clearer specifically what coordinate we return? E.g. to semantically differentiate between destination parameter and return object.

Reply:
In Rust I would do that. Make an extra variable too make it clearer what we are returning, the compiler would optimize it away anyways. But this is typescript, it won't optimize a variable assignment away, and this function is going to run on every element on every frame.

Reply:
Good point. I didn't consider that.